x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -309,14 +311,26 @@ class OutputService {
       );
     }
 
-    const updateData = { ...data };
+    const updateData: Nullable<Partial<Output>> = { ...data };
     const mergedType = data.type ?? originalOutput.type;
     const mergedIsDefault = data.is_default ?? originalOutput.is_default;
 
     if (mergedType === outputType.Logstash) {
       await validateLogstashOutputNotUsedInAPMPolicy(soClient, id, mergedIsDefault);
     }
 
+    // If the output type changed
+    if (data.type && data.type !== originalOutput.type) {
+      if (data.type === outputType.Logstash) {
+        // remove ES specific field
+        updateData.ca_trusted_fingerprint = null;
```

Comment:
wouldn't it be simpler to delete the property instead of setting it to `null`?

Reply:
To be able to delete it from the save object when we persit it we need to set it to null otherwise if the property is not here it will not update it.